OEP-6/OEP-6.mediawiki

```diff
@@ -0,0 +1,452 @@
+
+Smart code execution is two step process:
+
+# The transaction is send to the network.
+# The transaction is recorded to the blockchain at a later time.
+
+The <code>invoke</code> function will finish as soon as the transaction is sent to the network. To receive useful data from <code>method</code> execution it is necessary to start listening for events using <code>addEventListener</code>.
+
+* Rejects with '''NO_ACCOUNT''' in case the user is not signed in or has no accounts
+* Rejects with '''WRONG_ACCOUNT''' in case the <code>sender</code> is not one of own accounts
+* Rejects with '''MALFORMED_ACCOUNT''' in case the <code>account</code> is not a valid account
+* Rejects with '''MALFORMED_CONTRACT''' in case the <code>contract</code> is not hex encoded contract address
+* Rejects with '''MALFORMED_ADDRESS''' in case the <code>addresses</code> contains at least one address which is not a valid account nor identity
+
+
+====invokeRead====
+<pre>
+function invokeRead(contract: string, method: string, parameters: Parameter[]): Promise<any>
+</pre>
+
+Initiates a <code>method</code> call to a smart <code>contract</code> with supplied <code>parameters</code> in read only mode (preExec). This kind of method call does not write anything to blockchain and therefore does not need to be signed or paid for. It is processed without user interaction.
+
+* Rejects with '''MALFORMED_CONTRACT''' in case the <code>contract</code> is not hex encoded contract address
+
+
+====deploy====
+<pre>
+function deploy(account: string, code: string, name: string, version: string, author: string, 
+    email: string, description: string, needStorage: boolean, gasPrice: number, gasLimit: number): Promise<any>
+</pre>
+
+Initiates deployment of smart contract. The <code>code</code> parameter represents compiled code of smart contract for NEOVM. The <code>gasPrice</code> and <code>gasLimit</code> are hints for the dAPI provider and it should allow user to override those values before signing. The <code>account</code> will be set as payer of the transaction. It '''must''' be one of the accounts returned by <code>getOwnAccounts</code>
+
+* Rejects with '''NO_ACCOUNT''' in case the user is not signed in or has no accounts
+* Rejects with '''WRONG_ACCOUNT''' in case the <code>sender</code> is not one of own accounts
+* Rejects with '''MALFORMED_ACCOUNT''' in case the <code>account</code> is not a valid account
+
+
+====addEventListener====
+<pre>
+type EventListener = (data: any) => void;
+
+function addEventListener(listener: EventListener): void
+</pre>
+
+Starts listening for Notify events.
+
+====removeEventListener====
+<pre>
+function addEventListener(listener: EventListener): void
```

Comment:
Typo. Should be `removeEventListener`

Reply:
Yup. Typo. 
I am still deciding about the notify events and how to return the data to dApp.

Reply:
I decided to remove the listener functionality altogether and further explain the return value from invoke and invokeRead.